cql3/functions/functions.cc

```diff
@@ -360,6 +360,56 @@ functions::find(const function_name& name, const std::vector<data_type>& arg_typ
     return {};
 }
 
+// This function is created only for forward_service use, thus it only checks for
+// aggregate functions if no declared function was found.
+//
+// The reason for this function is, there is no serialization of `cql3::selection::selection`,
+// so functions lying underneath these selections has to be refound.
+//
+// Most of this code is copied from `functions::get()`, however `functions::get()` requires to 
+// mock or serialize expressions and `functions::find()` is not enough,
+// because it does not search for dynamic aggregate functions
+shared_ptr<function>
+functions::find_with_dynamic_aggr(const function_name &name, const std::vector<data_type> &arg_types) {
```

Comment:
Our coding style has `&` sticking to the type, not the argument.

I don't understand the function. What does "with_dynamic_aggr" mean? I could try to reverse-engineer the meaning from the code, but the function name should explain itself better.

Reply:
If it's copied from function::get(), maybe an extra parameter to function::get() is indicated, or sharing of the code with a helper.

Reply:
renamed and extracted the common part